resources/code-specific-lines.js

```diff
@@ -0,0 +1,47 @@
+Prism.hooks.add( 'complete', function ( env ) {
+    var showLineRanges = env.element.parentElement.getAttribute( 'data-show-lines' );
+
+    if (showLineRanges) {
+        var showLines = [];
+
+        showLineRanges.replace( /\s+/g, '' ).split( ',' ).filter( Boolean ).forEach( function ( currentRange ) {
+            if (currentRange.includes( '-' )) {
+                var range = currentRange.split( '-' );
+
+                if ( range[0] > range[1] ) {
+                    [range[0], range[1]] = [range[1], range[0]];
+                }
+
+                for ( var counter = range[0]; counter <= range[1]; counter++ ) {
+                    showLines.push( Number( counter ) );
+                }
+            }
+            else {
+                showLines.push( Number( currentRange ) );
+            }
+        });
```

Comment:
This could go into a nice pure-function with full unit-test coverage

Reply:
Modified